build.go

```diff
@@ -156,29 +128,26 @@ type spkParams struct {
 	srcDir string
 }
 
-func (p spkParams) spkBuild() int {
-	if p.dsm == 7 {
-		return 2000 + p.spkBuildBase
-	}
-	return p.spkBuildBase
+func (p spkParams) spkBuild(dsm int) int {
+	return 10*p.spkBuildBase + dsm
 }
 
-func (p spkParams) spkVersion() string {
-	return fmt.Sprintf("%v-%v", p.version, p.spkBuild())
+func (p spkParams) spkVersion(dsm int) string {
+	return fmt.Sprintf("%v-%v", p.version, p.spkBuild(dsm))
 }
 
 // filename returns the SPK's base filename.
-func (p spkParams) filename() string {
+func (p spkParams) filename(dsm int, arch string) string {
```

Comment:
use goArch or synoArch everywhere. "arch" by itself is ambiguous.

Reply:
ack